contracts/ibc-reflect-send/src/contract.rs

```diff
@@ -0,0 +1,649 @@
+    match msg {
+        HandleMsg::UpdateAdmin { admin } => handle_update_admin(deps, info, admin),
+        HandleMsg::SendMsgs { channel_id, msgs } => {
+            handle_send_msgs(deps, env, info, channel_id, msgs)
+        }
+        HandleMsg::CheckRemoteBalance { channel_id } => {
+            handle_check_remote_balance(deps, env, info, channel_id)
+        }
+        HandleMsg::SendFunds {
+            reflect_channel_id,
+            transfer_channel_id,
+        } => handle_send_funds(deps, env, info, reflect_channel_id, transfer_channel_id),
+    }
+}
+
+pub fn handle_update_admin(
+    deps: DepsMut,
+    info: MessageInfo,
+    new_admin: HumanAddr,
+) -> StdResult<HandleResponse> {
+    // auth check
+    let mut cfg = config(deps.storage).load()?;
+    if info.sender != cfg.admin {
+        return Err(StdError::generic_err("Only admin may set new admin"));
+    }
+    cfg.admin = new_admin;
+    config(deps.storage).save(&cfg)?;
+
+    Ok(HandleResponse {
+        messages: vec![],
+        attributes: vec![
+            attr("action", "handle_update_admin"),
+            attr("new_admin", cfg.admin),
+        ],
+        data: None,
+    })
+}
+
+pub fn handle_send_msgs(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    channel_id: String,
+    msgs: Vec<CosmosMsg>,
+) -> StdResult<HandleResponse> {
+    // auth check
+    let cfg = config(deps.storage).load()?;
+    if info.sender != cfg.admin {
+        return Err(StdError::generic_err("Only admin may send messages"));
+    }
+    // ensure the channel exists (not found if not registered)
+    accounts(deps.storage).load(channel_id.as_bytes())?;
+
+    // construct a packet to send
+    let packet = PacketMsg::Dispatch { msgs };
+    let msg = IbcMsg::SendPacket {
+        channel_id,
+        data: to_binary(&packet)?,
+        timeout_block: None,
+        timeout_timestamp: build_timeout(&env.block),
+    };
+
+    Ok(HandleResponse {
+        messages: vec![msg.into()],
+        attributes: vec![attr("action", "handle_send_msgs")],
+        data: None,
+    })
+}
+
+pub fn handle_check_remote_balance(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    channel_id: String,
+) -> StdResult<HandleResponse> {
+    // auth check
+    let cfg = config(deps.storage).load()?;
+    if info.sender != cfg.admin {
+        return Err(StdError::generic_err("Only admin may send messages"));
+    }
+    // ensure the channel exists (not found if not registered)
+    accounts(deps.storage).load(channel_id.as_bytes())?;
+
+    // construct a packet to send
+    let packet = PacketMsg::Balances {};
+    let msg = IbcMsg::SendPacket {
+        channel_id,
+        data: to_binary(&packet)?,
+        timeout_block: None,
+        timeout_timestamp: build_timeout(&env.block),
+    };
+
+    Ok(HandleResponse {
+        messages: vec![msg.into()],
+        attributes: vec![attr("action", "handle_check_remote_balance")],
+        data: None,
+    })
```

Comment:
Don't know much about IBC yet but, how the answer will be received / processed?

Reply:
Our of scope here... right now that is all "magic" that should be documented in the cosmos-sdk/wasmd.
I can add an IBC contract guide at some point. Ideally along with the test suite (mentioned above)